source/common/http/utility.cc

```diff
@@ -1006,7 +1006,7 @@ std::string Utility::PercentEncoding::encode(absl::string_view value, const size
     if (ch < ' ' || ch >= '~' || reserved_char_set.find(ch) != reserved_char_set.end()) {
       // For consistency, URI producers should use uppercase hexadecimal digits for all
       // percent-encodings. https://tools.ietf.org/html/rfc3986#section-2.1.
-      absl::StrAppend(&encoded, fmt::format("%{:02X}", ch));
+      absl::StrAppend(&encoded, fmt::format("%{:02X}", static_cast<const unsigned char&>(ch)));
```

Comment:
nit: would it better to convert `ch` to an unsigned char directly rather than convert the ref type by `static_cast<unsigned char>(ch)`

Reply:
I have not actually tried but if I converted ch to unsigned, will need to cast the const char in the if statement and when calling find(), so it's not as clean.